### PR TITLE
Fix: Fixed issue where thumbnails would sometimes fail to load in OneDrive

### LINKED ISSUE #14437: Bug: Thumbnails don't refresh in OneDrive after being cached
### Description

Thumbnails don't refresh in OneDrive after being cached

### Steps To Reproduce

1. Clear thumbnail cache in Windows
2. Open OneDrive location with images
3. Notice how only icons are displayed
4. Wait a second and then refresh the location, notice that the thumbnails are now displayed

### Requirements

- Refetch the thumbnail if the initial result was an icon

### Files Version

v3.1.1

### Windows Version

Windows 11

### Log File

NA

### PR DESCRIPTION
**Pull request by yaira2**
**Resolved / Related Issues**
- [ ] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #14437...

**Validation**
How did you test these changes?
- [ ] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?
   1. Confirm file icon is displayed and then replaced once the thumbnail is loaded

**Screenshots (optional)**
Add screenshots here.

### CODE REVIEW

File: src/Files.App/Data/Models/ItemViewModel.cs [hunk 1]
```diff
@@ -1064,6 +1065,11 @@ public async Task LoadExtendedItemPropertiesAsync(ListedItem item, uint thumbnai
 
 									SetFileTag(item);
 									wasSyncStatusLoaded = true;
+
+									var cancellationTokenSource = new CancellationTokenSource(3000);
+									// Loop until cached thumbnail is loaded or timeout is reached
+									while (!await LoadItemThumbnailAsync(item, thumbnailSize, matchingStorageFile))
+										cancellationTokenSource.Token.ThrowIfCancellationRequested();
```
Comment (hishitetsu) [inline src/Files.App/Data/Models/ItemViewModel.cs:1072]: `Task.Delay()` should be added to reduce load. I think 100msec is about right.

File: src/Files.App/Utils/Shell/Win32API.cs [hunk 2]
```diff
@@ -228,9 +228,31 @@ private class IconAndOverlayCacheEntry
 
 		private static readonly object _lock = new object();
 
-		public static (byte[]? icon, byte[]? overlay) GetFileIconAndOverlay(string path, int thumbnailSize, bool isFolder, bool getIconOnly, bool getOverlay = true, bool onlyGetOverlay = false)
+		/// <summary>
+		/// Returns an icon when a thumbnail isn't available or if getIconOnly is true
+		/// Returns an icon overlay when getOverlay is true
+		/// Returns a boolean indicating if the icon/thumbnail is cached
```
Comment (0x5bfa) [inline src/Files.App/Utils/Shell/Win32API.cs:234]:
```suggestion
		/// Returns an icon when a thumbnail isn't available or if getIconOnly is true.
		/// <br/>
		/// Returns an icon overlay when getOverlay is true.
		/// <br/>
		/// Returns a boolean indicating if the icon/thumbnail is cached.
```

File: src/Files.App/Utils/Shell/Win32API.cs [hunk 3]
```diff
@@ -228,9 +228,31 @@ private class IconAndOverlayCacheEntry
...
+		/// <returns></returns>
+		public static (byte[]? icon, byte[]? overlay, bool isIconCached) GetFileIconAndOverlay
+		(
+			string path,
+			int thumbnailSize,
+			bool isFolder,
+			bool getIconOnly,
+			bool getOverlay = true,
+			bool onlyGetOverlay = false
+		)
```
Comment (0x5bfa) [inline src/Files.App/Utils/Shell/Win32API.cs:251]:
```suggestion
		public static (byte[]? icon, byte[]? overlay, bool isIconCached) GetFileIconAndOverlay(
			string path,
			int thumbnailSize,
			bool isFolder,
			bool getIconOnly,
			bool getOverlay = true,
			bool onlyGetOverlay = false)
```

File: src/Files.App/Utils/Shell/Win32API.cs [hunk 4]
```diff
@@ -228,9 +228,31 @@ private class IconAndOverlayCacheEntry
...
+		/// Returns an icon overlay when getOverlay is true
+		/// Returns a boolean indicating if the icon/thumbnail is cached
+		/// </summary>
+		/// <param name="path"></param>
+		/// <param name="thumbnailSize"></param>
+		/// <param name="isFolder"></param>
+		/// <param name="getIconOnly"></param>
+		/// <param name="getOverlay"></param>
+		/// <param name="onlyGetOverlay"></param>
+		/// <returns></returns>
```
Comment (0x5bfa) [inline src/Files.App/Utils/Shell/Win32API.cs:242]: You can remove them if you don't add any comments.

**Review by 0x5bfa — COMMENTED** [review]
Pointed out some parts; otherwise LGTM!

File: src/Files.App/Data/Models/ItemViewModel.cs [hunk 5]
```diff
@@ -1064,6 +1065,14 @@ public async Task LoadExtendedItemPropertiesAsync(ListedItem item, uint thumbnai
 
 									SetFileTag(item);
 									wasSyncStatusLoaded = true;
+
+									var cancellationTokenSource = new CancellationTokenSource(3000);
+									// Loop until cached thumbnail is loaded or timeout is reached
+									while (!await LoadItemThumbnailAsync(item, thumbnailSize, matchingStorageFile))
+									{
+										await Task.Delay(100);
+										cancellationTokenSource.Token.ThrowIfCancellationRequested();
```
Comment (hishitetsu) [inline src/Files.App/Data/Models/ItemViewModel.cs:1074]:
```suggestion
										cancellationTokenSource.Token.ThrowIfCancellationRequested();
										await Task.Delay(100);
```
Better to check the timeout first.

**Review by hishitetsu — APPROVED** [review]
LGTM and works for me!

**Pull request merged**